### PR TITLE
MAINT: Bump pydata-sphinx-theme; redirect logo link from ToC to landing page

### LINKED ISSUE #570: Consider changing logo link in the documentation page from contents to index
#### Describe the issue linked to the documentation

Clicking the logo leads to contents instead of index. This is because `pydata-sphinx-theme` sets logo href to `master_doc` (which in `fairlearn` case is contents).

#### Suggest a potential alternative/fix
I'm not sure if linking the logo to `index` instead of `content` is really what `fairlearn` wants, but an extra opinion from `fairlearn` maintainers would be awesome.

I have created an issue in `pydata-sphinx-theme` and will mention this issue over there (https://github.com/pandas-dev/pydata-sphinx-theme/issues/245)

### PR DESCRIPTION
**Pull request by leonarduschen**
Closes #570 

WIP: Need to wait for `pydata-sphinx-theme` new release. Opening a PR now to keep me from forgetting.

### CODE REVIEW

**Comment by romanlutz** [comment]
Thanks @leonarduschen ! Can you take a look at the DCO issue in the checks? This may be helpful: https://fairlearn.github.io/master/contributor_guide/development_process.html#developer-certificate-of-origin

... but the help text when you click on "Details" in the checks list itself may be sufficient to fix it for this time.

**Comment by romanlutz** [comment]
> Closes #570
> 
> WIP: Need to wait for `pydata-sphinx-theme` new release. Opening a PR now to keep me from forgetting.

Two new releases out in the past week @leonarduschen in case you're still interested in making this change. https://pypi.org/project/pydata-sphinx-theme/0.4.3/#history

**Comment by leonarduschen** [comment]
The changes are not released yet though - only a small subset of all merges were released.

**Comment by romanlutz** [comment]
> The changes are not released yet though - only a small subset of all merges were released.

Ah! Apologies!

**Comment by romanlutz** [comment]
> > The changes are not released yet though - only a small subset of all merges were released.
> 
> Ah! Apologies!

I believe v0.5.0 includes it now :-)
https://github.com/pydata/pydata-sphinx-theme/releases

Do you mind bumping the version in the requirements-dev.txt file to `pydata-sphinx-theme>=0.5.0`?

We're getting rid of the DCO requirement shortly since it's quite painful to make it work. You can read up on what it means [here](https://fairlearn.org/v0.6.0/contributor_guide/development_process.html#developer-certificate-of-origin) but I'm happy to complete the PR if you acknowledge in a comment that you agree to the DCO. It would definitely save some time...

Thanks for getting this into pydata-sphinx-theme and putting up this PR!

**Comment by leonarduschen** [comment]
All done!

File: docs/conf.py [hunk 1]
```diff
@@ -107,6 +107,7 @@ def check_if_v046():
 # further.  For a list of options available for each theme, see the
 # documentation.
 html_theme_options = {
+    "logo_link": "index",
```
Comment (leonarduschen) [inline docs/conf.py:110]: I noticed this TODO comment - shall we link the logo to one of these links instead?

File: docs/conf.py [hunk 2]
```diff
@@ -107,6 +107,7 @@ def check_if_v046():
...
     # TODO: fork the pydata-sphinx-theme to integrate these with logo
```
Comment (leonarduschen) [inline docs/conf.py:111]: Wrong reference, I meant this comment

**Comment by riedgar-ms** [comment]
Sorry, could you clarify what's supposed to be changing? I'm looking at the docs produced in your build, and it appears that clicking on the logo in the top left of the page takes the user back to the table of contents. The same thing happens with the fairlearn.org docs.

**Comment by leonarduschen** [comment]
> Sorry, could you clarify what's supposed to be changing? I'm looking at the docs produced in your build, and it appears that clicking on the logo in the top left of the page takes the user back to the table of contents. The same thing happens with the fairlearn.org docs.

Yeah my bad - I didn't realize the table of contents now is exactly index.html (I think it wasn't when the PR was first created?). Where should we link the logo to ideally? If it is table of contents is the intended link then I think this PR is not needed.

**Comment by riedgar-ms** [comment]
Thanks for double checking - I was worried that I'd completely misunderstood what was supposed to be happening.

**Comment by romanlutz** [comment]
> > Sorry, could you clarify what's supposed to be changing? I'm looking at the docs produced in your build, and it appears that clicking on the logo in the top left of the page takes the user back to the table of contents. The same thing happens with the fairlearn.org docs.
> 
> Yeah my bad - I didn't realize the table of contents now is exactly index.html (I think it wasn't when the PR was first created?). Where should we link the logo to ideally? If it is table of contents is the intended link then I think this PR is not needed.

Clicking the logo should take you to the landing page (`fairlearn.org`) rather than the TOC. That's what you were trying to fix if I understand correctly. That's also what the expected behavior is in my mind. :-)

**Comment by leonarduschen** [comment]
@romanlutz @riedgar-ms changed to fairlearn.org

**Review by romanlutz — APPROVED** [review]
This is great! I checked it out through CircleCI and it looks great. We just need a second sign-off @riedgar-ms @adrinjalali @lejit perhaps one of you has thoughts, otherwise I'd love to merge this.
Thanks @leonarduschen !

**Review by riedgar-ms — APPROVED** [review]
Thanks!

**Comment by romanlutz** [comment]
Awesome! We have all the necessary sign-offs, so I'll complete it tomorrow (Thursday) unless there are comments to the contrary.

**Comment by leonarduschen** [comment]
ping

**Comment by romanlutz** [comment]
Thanks for the reminder. I was out a few days. Merged it now. [I keep using "complete" since a different tool used that term....]

**Pull request merged**